sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/utils/CalciteUtils.java

```diff
@@ -156,8 +156,26 @@ public static boolean isStringType(FieldType fieldType) {
           .put(TIMESTAMP_WITH_LOCAL_TZ, SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
           .build();
 
-  private static final BiMap<SqlTypeName, FieldType> CALCITE_TO_BEAM_TYPE_MAPPING =
-      BEAM_TO_CALCITE_TYPE_MAPPING.inverse();
+  private static final ImmutableMap<SqlTypeName, FieldType> CALCITE_TO_BEAM_TYPE_MAPPING =
+      ImmutableMap.<SqlTypeName, FieldType>builder()
+          .put(SqlTypeName.TINYINT, TINY_INT)
+          .put(SqlTypeName.SMALLINT, SMALL_INT)
+          .put(SqlTypeName.INTEGER, INTEGER)
+          .put(SqlTypeName.BIGINT, BIG_INT)
+          .put(SqlTypeName.FLOAT, FLOAT)
+          .put(SqlTypeName.DOUBLE, DOUBLE)
+          .put(SqlTypeName.DECIMAL, DECIMAL)
+          .put(SqlTypeName.BOOLEAN, BOOLEAN)
+          .put(SqlTypeName.VARBINARY, VARBINARY)
+          .put(SqlTypeName.BINARY, VARBINARY)
```

Comment:
Both BINARY and VARBINARY should be mapped to `VARBINARY`, which  is FieldType.BYTES.

Reply:
BINARY might generate from `select b'test_string'`.

Reply:
OK. It is fine and expected that the mapping might not be invertible.

We want users to be able to feed schema PCollection in to SQL, but also compile SQL down to schema. But they don't have to be an exact match. In fact, I think Beam schema should probably reduce to only very simple types and the rest should be logical types that SQL defines. TODO later.

You will need to replace `VARBINARY` with `BYTES` on the right hand side, no?

Reply:
Agree on the logical types and leave it for future work for now.

It's a legacy code that defines `public static final FieldType VARBINARY = FieldType.BYTES;`. I am reusing this style.

Reply:
Ah I see. I don't like it but I accept it :-)